src/versions.rs

```diff
@@ -16,32 +16,180 @@
+/// version is found for the corresponding Polkadot version. In case of any error during the
+/// fetching or parsing process, an error is returned.
+///
+/// # Errors
+///
+/// This function returns an error if there is any issue with the HTTP request, response parsing,
+/// or if the required fields are not found in the `Cargo.dev.toml` file.
+///
+/// # Examples
+///
+/// ```
+/// #[tokio::main]
+/// async fn main() {
+///     let base_url = "https://raw.githubusercontent.com";
+///     let version = "1.12.0";
+///     match get_orml_crates_and_version(base_url, version).await {
+///         Ok(Some(orml_toml)) => println!("ORML crates: {:?}", orml_toml),
+///         Ok(None) => println!("No matching ORML version found."),
+///         Err(e) => println!("Error fetching ORML crates: {}", e),
+///     }
+/// }
+/// ```
+pub async fn get_orml_crates_and_version(
+    base_url: &str,
+    version: &str,
+) -> Result<Option<OrmlToml>, Box<dyn std::error::Error>> {
+    if get_release_branches_versions(Repository::Orml)
+        .await?
+        .contains(&version.to_string())
+    {
+        let version_url = format!(
+            "{}/open-web3-stack/open-runtime-module-library/polkadot-v{}/Cargo.dev.toml",
+            base_url, version
+        );
+        let response = reqwest::Client::new()
+            .get(&version_url)
+            .header("User-Agent", "reqwest")
+            .header("Accept", "application/vnd.github.v3+json")
+            .send()
+            .await?;
+
+        let content = response.text().await?;
+
+        let orml_workspace_members = toml::from_str::<OrmlToml>(&content)
+            .map_err(|_| return "Error Parsing ORML TOML. Required Fields not Found")?;
+        Ok(Some(orml_workspace_members))
+    } else {
+        log::error!(
+            "No matching ORML release version found for corresponding polkadot-sdk version."
+        );
+        Ok(None)
+    }
+}
+
+/// Includes ORML crates in the version mapping.
+///
+/// This function updates a given version mapping (`BTreeMap`) by adding the versions of ORML 
+/// crates obtained from a `OrmlToml` instance. It prefixes each crate name with "orml-" and 
+/// inserts the corresponding version into the map. If the `orml_crates_version` is `None`, 
+/// the function does nothing.
+///
+/// # Arguments
+///
+/// * `crates_versions` - A mutable reference to a `BTreeMap` where the original polkadot-sdk 
+///   crate names and versions are stored.
+/// * `orml_crates_version` - An `Option<OrmlToml>` that may contain the ORML crates and their
+///   versions.
+///
+/// # Examples
+///
+/// ```
+/// let mut version_map: BTreeMap<String, String> = BTreeMap::new();
+/// include_orml_crates_in_version_mapping(&mut version_map, Some(orml_toml));
+/// ```
+pub fn include_orml_crates_in_version_mapping(
+    crates_versions: &mut BTreeMap<String, String>,
+    orml_crates_version: Option<OrmlToml>,
+) {
+    if let Some(orml_toml) = orml_crates_version {
```

Comment:
A function accepting an `Option` and doing nothing in the `None` case is a bit odd.

Reply:
The function actually shouldn't do anything if it's None. It's the case where if --orml flag is supplied but there is no release version for the version supplied, the function shouldn't do anything because in that case it's None. So rather than checking if it's `is_some()` or `is_none()` I am just calling the function. If it's None then nothing happens.

Reply:
What i meant is that the function should just be called in that case so that it can accept a `OrmlToml`.